File: ocmsdk/src/main/java/com/gigigo/orchextra/core/data/api/mappers/elements/ApiElementCustomPropertyMapper.java
package com.gigigo.orchextra.core.data.api.mappers.elements;

import com.gigigo.ggglib.mappers.ExternalClassToModelMapper;
import com.gigigo.orchextra.core.data.api.dto.elements.ApiElementCustomProperty;
import com.gigigo.orchextra.core.domain.entities.elements.ElementCustomProperty;

public class ApiElementCustomPropertyMapper
    implements ExternalClassToModelMapper<ApiElementCustomProperty, ElementCustomProperty> {

  @Override public ElementCustomProperty externalClassToModel(ApiElementCustomProperty data) {
    ElementCustomProperty model = new ElementCustomProperty();

    model.setCustomProperties(data.getCustomProperties());

    return model;
  }
}

File: ocmsdk/src/main/java/com/gigigo/orchextra/core/data/api/mappers/elements/ApiElementCustomPropertyMapper.kt
package com.gigigo.orchextra.core.data.api.mappers.elements

import com.gigigo.ggglib.mappers.ExternalClassToModelMapper
import com.gigigo.orchextra.core.data.api.dto.elements.ApiElementCustomProperty
import com.gigigo.orchextra.core.domain.entities.elements.ElementCustomProperties

class ApiElementCustomPropertyMapper : ExternalClassToModelMapper<ApiElementCustomProperty, ElementCustomProperties> {

  override fun externalClassToModel(data: ApiElementCustomProperty): ElementCustomProperties {
    val model = ElementCustomProperties()

    model.properties = data.properties ?: HashMap()

    return model
  }
}
